blocktopmenu.php

```diff
@@ -1358,4 +1371,104 @@ public function renderList()
+
+        if (isset($category['children']) && !empty($category['children'])) {
+            foreach ($category['children'] as $child) {
+                if ($childNode = $this->createCategoryNode($child)) {
+                    $categoryNode->addChild($childNode);
+                }
+            }
+            if ($addThumbnails && $category['level_depth'] > 1) {
+                $images = array();
+                $files = scandir(_PS_CAT_IMG_DIR_);
+                foreach ($files as $file) {
+                    if (preg_match('/^'.$category['id_category'].'-([0-9])?_thumb.jpg/i', $file) === 1) {
+                        $images[] = array(
+                            'src' => $this->context->link->getMediaLink(_THEME_CAT_DIR_.$file),
+                            'alt' => Tools::SafeOutput($category['name']),
+                            'title' => Tools::SafeOutput($category['name']),
+                        );
+                    }
+                }
+                if (!empty($images)) {
+                    $thumbnailsNode = $this->createNode(BlocktopmenuNode::TYPE_CATEGORY_THUMBNAILS, array(
+                        'images' => $images,
+                    ));
+                    $categoryNode->addChild($childNode);
+                }
+            }
+        }
+
+        return $categoryNode;
+    }
+
+    private function createCmsCategoryNode(CMSCategory $cmsCategory)
+    {
+        $cmsCategoryNode = $this->createNode(BlocktopmenuNode::TYPE_CMS_CATEGORY, array(
+            'id' => $cmsCategory->id,
+            'link' => Tools::HtmlEntitiesUTF8($cmsCategory->getLink()),
+            'name' => $cmsCategory->name,
+        ));
+
+        $id_lang = Context::getContext()->language->id;
+
+        $categories = $this->getCMSCategories(false, $cmsCategory->id, $id_lang);
+        $pages = $this->getCMSPages($cmsCategory->id);
+
+        if (count($categories) || count($pages)) {
+            foreach ($categories as $category) {
+                $cmsSubCategory = new CMSCategory($category['id_cms_category'], $id_lang);
+                $cmsCategoryNode->addChild($this->createCmsCategoryNode($cmsSubCategory));
+            }
+            foreach ($pages as $page) {
+                $cms = new CMS($page['id_cms'], $id_lang);
+                $links = $cms->getLinks($id_lang, array($cms->id));
+                $selected = $this->page_name == 'cms' && Tools::getValue('id_cms') == $page['id_cms'];
+                $cmsNode = $this->createNode(BlocktopmenuNode::TYPE_CMS, array(
+                    'id' => $cms->id,
+                    'link' => Tools::HtmlEntitiesUTF8($links[0]['link']),
+                    'name' => $cms->meta_title,
+                ));
+                $cmsNode->setSelected($selected);
+                $cmsCategoryNode->addChild($cmsNode);
+            }
+        }
+
+        return $cmsCategoryNode;
+    }
+
+    private static function renderNode(BlocktopmenuNode $node)
```

Comment:
Because it is only here for backward compatibility: the module still assigns the `$MENU` variable with HTML code. Future versions should not generate the HTML anymore and this method should be deprecated ASAP. 
If you want to change the HTML code, you should do it in the view, not in the controller.